tests/unit/customizations/ec2instanceconnect/test_websocket.py

```diff
@@ -292,6 +292,41 @@ def test_connect_with_proxy(
 
         assert mock_on_connection_event.wait.called
 
+    @patch.object(websocket, "connect")
+    @patch.object(websocket, "create_handshake_request")
+    @mock.patch.dict(os.environ, {"HTTPS_PROXY": "http://user1:pass1@localhost:8989", "NO_PROXY": ""})
```

Comment:
Let's remove the `NO_PROXY` env var in patched environment variables to make sure the test case represents the bug that this fixes, which is when `HTTPS_PROXY` but `NO_PROXY` is not present as an environment variable.